lib/agent.js

```diff
@@ -143,12 +144,48 @@ class Agent {
             return
         }
         this.updating = true
+
+        /** `forgeSnapshot` will be set to the current snapshot in the forge platform *if required* */
+        let forgeSnapshot = null
+
+        // first, check if the new state indicates a change of operation mode from the current mode
+        // When changing from developer mode to autonomous mode, we need to check if the flows/modules
+        // for Node-RED were changed vs the current snapshot on the forge platform.
+        // If they differ, we flag that a reload of the snapshot is required.
         if (newState !== null && newState.mode && newState.mode !== this.currentMode) {
-            // Device mode has been changed
-            // update the local state before processing the new state options
-            this.currentMode = newState.mode
-            await this.saveProject()
+            ['developer', 'autonomous'].includes(newState.mode) || (newState.mode = 'autonomous')
+            const modeChanged = this.currentMode !== newState.mode
+            if (modeChanged) {
+                this.currentMode = newState.mode
+                if (newState.mode === 'developer') {
+                    info('Switching on developer mode')
+                    await this.saveProject()
+                } else {
+                    let _launcher = this.launcher
+                    if (!_launcher) {
+                        // create a temporary launcher to read the current snapshot on disk
+                        _launcher = Launcher.newLauncher(this.config, this.currentProject, this.currentSnapshot, this.currentSettings, this.currentMode)
+                    }
+                    const modules = (await _launcher.readPackage())?.modules
+                    const flows = await _launcher.readFlow()
+                    const diskSnapshot = { flows, modules }
+                    forgeSnapshot = await this.httpClient.getSnapshot()
+                    newState.reloadSnapshot = utils.compareNodeRedData(forgeSnapshot, diskSnapshot) === false
+                    if (newState.reloadSnapshot) {
+                        info('Current Node-RED files differ from the Platform snapshot, a reload will be requested')
```

Comment:
```suggestion
                        info('Restoring target snapshot')
```

Reply:
Counter suggestion:

```suggestion
                        info('Node-RED differs from the Platform snapshot, a reload will be requested')
```

I prefer to be more explicit in the intent of the message Nick.  This log not made due to the action but rather the fact it will occur later (other intermediate logs may yet occur, confusing the overall log output)

Reply:
To me, 'reload will be requested' it feels too passive. Yes the agent will request the snapshot from the platform, but the platform isn't going to say no.

How about:

```suggestion
                        info('Local flows have changed. Restoring current snapshot')
```